project_forecast_line/models/project_task.py

```diff
@@ -18,9 +18,9 @@ class ProjectTask(models.Model):
     def create(self, vals_list):
         # compatibility with fields from project_enterprise
         for vals in vals_list:
-            if "planned_date_begin" in vals:
+            if "planned_date_begin" in vals and vals.get("planned_date_begin"):
```

Comment:
here if we have empty values of `planned_date_begin` or `planned_date_end` on create, then we won't compute forecast as `forecast_date_planned_start` and `forecast_date_planned_end` will be empty too

Reply:
side note: planned_date_begin and planned_date_end are fields introduced in a enterprise module. We cannot depend on them being set on the CE version, this is a hack to avoid overloading the form view in a complex way or introducing a glue module depending on Odoo EE

your fix is correct

Reply:
```suggestion
            if vals.get("planned_date_begin"):
```
is shorter and does the same

Reply:
thanks for explanation